lava-api/src/job.rs

```diff
@@ -538,6 +539,32 @@ pub async fn cancel_job(lava: &Lava, id: i64) -> Result<(), CancellationError> {
     }
 }
 
+#[derive(Error, Debug)]
+pub enum ResultsError {
+    #[error("Request failed {0}")]
+    Request(#[from] reqwest::Error),
+    #[error("Unexpected reply: {0}")]
+    UnexpectedReply(reqwest::StatusCode),
+}
+
+pub async fn job_results_as_junit(lava: &Lava, id: i64) -> Result<Bytes, ResultsError> {
```

Comment:
Rather then returning Bytes shouldn't this return a stream instead? As we're not parsing it might make sense to leave it to the user whether to fully read it into memory or not